executor/ddl.go

```diff
@@ -392,24 +409,28 @@ func (e *DDLExec) getRecoverTableByTableName(s *ast.RecoverTableStmt, t *meta.Me
 	if err != nil {
 		return nil, nil, err
 	}
-	schemaName := s.Table.Schema.L
+	schemaName := tableName.Schema.L
 	if schemaName == "" {
 		schemaName = e.ctx.GetSessionVars().CurrentDB
 	}
 	if schemaName == "" {
 		return nil, nil, errors.Trace(core.ErrNoDB)
 	}
+	dom := domain.GetDomain(e.ctx)
 	// TODO: only search recent `e.JobNum` DDL jobs.
 	for i := len(jobs) - 1; i > 0; i-- {
 		job = jobs[i]
-		if job.Type != model.ActionDropTable {
+		if job.Type != model.ActionDropTable && job.Type != model.ActionTruncateTable {
 			continue
 		}
 		// Check GC safe point for getting snapshot infoSchema.
 		err = gcutil.ValidateSnapshotWithGCSafePoint(job.StartTS, gcSafePoint)
 		if err != nil {
 			return nil, nil, err
 		}
+		if len(ts) != 0 && ts != model.TSConvert2Time(job.StartTS).String() {
+			continue
```

Comment:
Why `flushback` syntax requires the `ts` exactly the same with the job's startTS?

Reply:
It is the PM demand...

Reply:
OK